backend/src/main/java/de/neuefische/backend/service/EventHubUserDetailService.java

```diff
@@ -36,14 +41,38 @@ public EventHubUserDTO saveUser(UserLoginAndRegistrationDTO user) {
                 .id(generateUUIDService.generateUUID())
                 .username(user.getUsername())
                 .password(passwordEncoder.encode(user.getPassword()))
+                .preferredCategories(new ArrayList<>())
                 .build();
 
-        if(userRepository.findEventHubUserByUsername(newUser.getUsername()).isPresent()) {
+        if (userRepository.findEventHubUserByUsername(newUser.getUsername()).isPresent()) {
             throw new IllegalArgumentException("User with username " + user.getUsername() + " already exists");
         }
 
         EventHubUser temp = userRepository.save(newUser.withRoles(Collections.singletonList(new SimpleGrantedAuthority("user"))).withPassword(passwordEncoder.encode(user.getPassword())));
         List<String> roles = temp.getRoles().stream().map(SimpleGrantedAuthority::toString).toList();
-        return new EventHubUserDTO(temp.getId(), temp.getUsername(), roles);
+        return new EventHubUserDTO(temp.getId(), temp.getUsername(), roles, temp.getPreferredCategories());
     }
+
+    public List<String> getUserPreferredCategories(String username) {
+        EventHubUser user = userRepository.findEventHubUserByUsername(username)
```

Comment:
user is similiar to temp. While it says it is a user it does not state what kind of user or what this user does. It is the currently logged in user in the event hub?

Reply:
In this case I disagree with you. I think that the connection between the name of the method and the term user provides sufficient information here. It is about the user whose preferred categories i want to retrieve. To put this information once more into the variable name would make the name too long and would be redundant. The method can be used regardless of a login status.